plugins/services/src/js/components/ServicesQuotaOverviewDetail.tsx

```diff
@@ -0,0 +1,173 @@
+import { Trans } from "@lingui/macro";
+import React from "react";
+import { of, Observable } from "rxjs";
+import {
+  map,
+  catchError,
+  distinctUntilChanged,
+  switchMap
+} from "rxjs/operators";
+import { componentFromStream } from "@dcos/data-service";
+import gql from "graphql-tag";
+import { DataLayer, DataLayerType } from "@extension-kid/data-layer";
+
+import Loader from "#SRC/js/components/Loader";
+import container from "#SRC/js/container";
+import { findNestedPropertyInObject } from "#SRC/js/utils/Util";
+import ProgressBar from "#SRC/js/components/ProgressBar";
+import * as ResourcesUtil from "#SRC/js/utils/ResourcesUtil";
+
+import { ServiceGroup, QuotaResources } from "../types/ServiceGroup";
+
+export interface ServicesQuotaOverviewDetailProps {
+  id: string;
+}
+
+function getQuotaPercentage(
+  group: ServiceGroup,
+  resource: string
+): number | null {
+  const resourceQuota: QuotaResources | undefined = findNestedPropertyInObject(
+    group.quota,
+    resource
+  );
+  if (!resourceQuota || !resourceQuota.limit) {
+    return null;
+  }
+
+  if (!resourceQuota.consumed) {
+    return 0;
+  }
+
+  return (resourceQuota.consumed / resourceQuota.limit) * 100;
+}
+
+function getQuotaConsumedOfLimit(
+  group: ServiceGroup,
+  resource: string
+): React.ReactNode | null {
+  const resourceQuota: QuotaResources | undefined = findNestedPropertyInObject(
+    group.quota,
+    resource
+  );
+
+  if (!resourceQuota || !resourceQuota.limit) {
+    return null;
+  }
+
+  if (resource === "cpus" || resource === "gpus") {
+    return (
+      <Trans>
+        {resourceQuota.consumed} of {resourceQuota.limit} Cores
+      </Trans>
+    );
+  }
+  return (
+    <Trans>
+      {resourceQuota.consumed} of {resourceQuota.limit} MiB
+    </Trans>
+  );
+}
+
+const titleFor = (resource: string): string => {
+  switch (resource) {
+    case "cpus":
+      return "CPUs";
+    case "gpus":
+      return "GPUs";
+    case "disk":
+      return "Disk";
+    case "memory":
+      return "Memory";
+  }
+  return "";
+};
+
+function getCard(group: ServiceGroup, resource: string): React.ReactNode {
```

Comment:
I feel like this should be a component outside of this file.

Reply:
But I'm ok with refactoring this later.

Reply:
I agree with both of Rodney's comments